File: HandSock-Server/src/main/java/com/server/handsock/websocket/SocketIOListener.java
package com.server.handsock.websocket;

import com.corundumstudio.socketio.*;
import org.springframework.stereotype.Service;
import com.server.handsock.websocket.listener.*;
import com.server.handsock.console.ConsolePrints;
import com.server.handsock.service.ClientService;
import com.server.handsock.websocket.eventer.OnlineEvent;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Map;

@Service
public class SocketIOListener {

    private final Map<String, Object> service;
    private final Map<String, Object> listener;

    @Autowired
    public SocketIOListener(Map<String, Object> service, Map<String, Object> listener) {
        this.service = service;
        this.listener = listener;
    }

    public void addServerEventListener(SocketIOServer server) {
        OnlineEvent onlineEvent = new OnlineEvent();

        server.addConnectListener(client -> {
            ClientService clientService = new ClientService(service);
            onlineEvent.sendUserConnect(server, client, clientService.getRemoteAddress(client));
        });

        server.addDisconnectListener(client -> {
            ClientService clientService = new ClientService(service);
            onlineEvent.sendUserDisconnect(server, client, clientService.getRemoteAddress(client));
        });

        server.addEventListener("[ONLINE:LOGIN]", Map.class, (client, data, ackSender) -> {
            ackSender.sendAckData(Map.of("code", 200));
            @SuppressWarnings("unchecked")
            Map<String, Object> typedData = (Map<String, Object>) data;
            onlineEvent.sendUserOnlineLogin(server, client, typedData);
        });

        try {
            CheckListener checkListener = (CheckListener) listener.get("checkListener");
            checkListener.addEventListener(server, service);

            SendListener sendListener = (SendListener) listener.get("sendListener");
            sendListener.addEventListener(server, service);

            EditListener editListener = (EditListener) listener.get("editListener");
            editListener.addEventListener(server, service);

            AdminListener adminListener = (AdminListener) listener.get("adminListener");
            adminListener.addEventListener(server, service);

            UserListener userListener = (UserListener) listener.get("userListener");
            userListener.addEventListener(server, service, onlineEvent);

            SystemListener systemListener = (SystemListener) listener.get("systemListener");
            systemListener.addEventListener(server, service);

            ClientListener clientListener = (ClientListener) listener.get("clientListener");
            clientListener.addEventListener(server, service);

            SearchListener searchListener = (SearchListener) listener.get("searchListener");
            searchListener.addEventListener(server, service);

            ReportListener reportListener = (ReportListener) listener.get("reportListener");
            reportListener.addEventListener(server, service);
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
        }
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/socket/SocketIOListener.kt
package com.server.handsock.socket

import com.corundumstudio.socketio.AckRequest
import com.corundumstudio.socketio.SocketIOClient
import com.corundumstudio.socketio.SocketIOServer
import com.server.handsock.common.utils.ConsoleUtils
import com.server.handsock.common.utils.HandUtils
import com.server.handsock.socket.eventer.OnlineEvent
import com.server.handsock.socket.listener.*

class SocketIOListener(
    private val onlineEvent: OnlineEvent,
    private val authListener: AuthListener,
    private val forumListener: ForumListener,
    private val adminListener: AdminListener,
    private val checkListener: CheckListener,
    private val updateListener: UpdateListener,
    private val clientListener: ClientListener,
    private val searchListener: SearchListener,
    private val configListener: ConfigListener,
    private val reportListener: ReportListener,
    private val sendingListener: SendingListener,
    private val commentListener: CommentListener
) {
    fun addServerEventListener(server: SocketIOServer) {
        server.addConnectListener {
            onlineEvent.sendUserConnect(
                client = it,
                server = server
            )
        }

        server.addDisconnectListener {
            onlineEvent.sendUserDisconnect(
                client = it,
                server = server
            )
        }

        server.addEventListener("[CLIENT:PING]", Map::class.java) { _: SocketIOClient?, _: Map<*, *>?, ackSender: AckRequest ->
            ackSender.sendAckData(HandUtils.handleResultByCode(200, null, "心跳正常"))
        }

        server.addEventListener("[ONLINE:LOGIN]", Map::class.java) { client: SocketIOClient?, data: Map<*, *>?, ackSender: AckRequest ->
            ackSender.sendAckData(java.util.Map.of("code", 200))
            @Suppress("UNCHECKED_CAST")
            val typedData = data as Map<String, Any>?
            onlineEvent.sendUserOnlineLogin(server, client!!, typedData!!)
        }

        try {
            adminListener.addEventListener()
            forumListener.addEventListener()
            clientListener.addEventListener()
            searchListener.addEventListener()
            authListener.addEventListener()
            configListener.addEventListener()
            updateListener.addEventListener()
            reportListener.addEventListener()
            sendingListener.addEventListener()
            commentListener.addEventListener()
            checkListener.addEventListener(HandUtils)
        } catch (e: Exception) {
            ConsoleUtils.printErrorLog(e)
        }
    }
}